docs/docs/tutorials/debugging.md

```diff
@@ -0,0 +1,89 @@
+# :material-bug: Hardware Debugging
+
+## Background
+
+For debugging our boards, we use OpenOCD and GDB.
+
+[OpenOCD (Open On-Chip Debugger)](https://openocd.org/) is a tool that enables on-chip debugging, flash programming, and testing for microcontrollers. It will start a server that listens for commands from various sources such as GDB.
+
+[GDB (GNU Debugger)](https://www.sourceware.org/gdb/) is a debugging tool that enables inspection, modification, and control of program execution on microcontrollers.
+
+## Installation
+
+Use the following command to install OpenOCD:
+
+=== "Windows"
+    ```text
+    <!-- TODO: Rida documentation for OpenOCD -->
+    ```
+
+=== "Linux / Mac"
+    ```text
+    brew install openocd
+    ```
+
+It is likely the ARM GNU Toolchain containing GDB is already installed if you have followed the [Firmware Development Setup](../firmware/dev-setup.md) guide. Refer to this guide if it is not installed.
+
+### Verify Installation
+
+Check that both programs are installed and have an acceptable version.
+
+Do not copy the `# version comments`.
+
+```text
+openocd --version               # >= 0.12.0
+arm-none-eabi-gdb --version     # >= 13.0
+```
+
+## Usage
+
+### Debugging
+
+To debug a program, the OpenOCD server must be started and GDB connected to it.
+
+You must have a compiled binary file to be able to load it onto the board for debugging. If you have not yet compiled your program, refer to the [Compiling a Project](../firmware/compile-project.md) guide.
```

Comment:
can these instructions be a list? I think it would be shorter and more clear where the reader needs to take action

1. Build your project for the `stm32f767` platform. See [Compiling a project]() for more details.
    
    ```bash
    make PROJECT=Demo/Blink PLATFORM=stm32f767 build
    ```

    You do not need to flash it to the board; GDB will handle that.

2. Start the OpenOCD server.

    ```bash
    make PROJECT=Demo/Blink PLATFORM=stm32f767 debug-openocd
    ```

    Wait until it says `Info : Listening on port 3333 for gdb connections`

3. __In a separate terminal__, start GDB

    ```bash
    make PROJECT=Demo/Blink PLATFORM=stm32f767 debug-gdb
    ```

4. Start debugging in the GDB terminal. See the list of common commands below.

Reply:
This lets you remove the warning, since the chronology is baked in. Also simplifies the compiling sentence since you just provide an implementation and link for more details. The whole section (from ##Usage to ##Common commands) can be replaced by these steps